client/src/js/app.js

```diff
@@ -1020,6 +1033,23 @@ function startupConfig($rootScope, $state, SessionService, amMoment, Notify) {
       event.preventDefault();
       Notify.warn('AUTH.CANNOT_RETURN_TO_LOGIN');
     }
+
+    var paths = SessionService.path;
+    var authorized = 0;
+    var currentPath = $location.$$path;
+
+    if(paths && currentPath !== '/' && currentPath !=='/settings' && currentPath !== '/login' && currentPath !== '/error404'){
+      paths.forEach(function (data) {
```

Comment:
This could be done much more clearly written using [Array.prototype.some](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/some).

Like this:
```js
var authorized = paths.some(function (data) {
  return currentPath.indexOf(data.path) === 0;
});

if (!authorized) { $location.path('/error403'); }
```